internal/authz/authz.go

```diff
@@ -166,7 +169,54 @@ func (a *Authorizer) AuthorizeCreateStore(ctx context.Context) error {
 		return err
 	}
 
-	return a.individualAuthorize(ctx, claims.ClientID, relation, System, &openfgav1.ContextualTupleKeys{})
+	return a.individualAuthorize(ctx, claims.ClientID, relation, SystemIDType, &openfgav1.ContextualTupleKeys{})
+}
+
+// AuthorizeListStores checks if the user has access to list stores.
+func (a *Authorizer) AuthorizeListStores(ctx context.Context) error {
+	claims, err := checkAuthClaims(ctx)
+	if err != nil {
+		return err
+	}
+
+	relation, err := a.getRelation(ListStores)
+	if err != nil {
+		return err
+	}
+
+	return a.individualAuthorize(ctx, claims.ClientID, relation, SystemIDType, &openfgav1.ContextualTupleKeys{})
+}
+
+// ListAuthorizedStores returns the list of stores that the user has access to.
+func (a *Authorizer) ListAuthorizedStores(ctx context.Context) ([]string, error) {
+	claims, err := checkAuthClaims(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &openfgav1.ListObjectsRequest{
+		StoreId:              a.config.StoreID,
+		AuthorizationModelId: a.config.ModelID,
+		User:                 ClientIDType(claims.ClientID).String(),
+		Relation:             CanCallGetStore,
+		Type:                 StoreType,
+	}
+
+	// Disable authz check for the list objects request.
+	ctx = authclaims.ContextWithSkipAuthzCheck(ctx, true)
+	resp, err := a.server.ListObjects(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	storeIDs := make([]string, len(resp.GetObjects()))
+	re := regexp.MustCompile(`^.*:(.*)`)
+	for i, store := range resp.GetObjects() {
+		id := re.FindStringSubmatch(store)
```

Comment:
Suggestion for a future PR: when you pull `main` into your branch, you can get rid of the regex matching and use the helper method `ToUserParts`: https://github.com/openfga/openfga/blob/f3be13c8969feffa443eed902b3dc377a8b9b8f4/pkg/tuple/tuple_test.go#L579-L599

Reply:
I am not sure I like that we're using a regex - 

can we do something like this instead:
```go
for index, store := range resp.GetObjects() {
storeIDs[index] = strings.TrimPrefix(store, `store:`)
```